plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotConnectorSmokeTest.java

```diff
@@ -969,6 +989,146 @@ public void testIntegerType()
+                .skippingTypesCheck()
+                .matches(otherVendors)
+                .isFullyPushedDown();
+
+        // not equals false case
+        assertThat(query("SELECT vendor FROM " + JSON_TABLE + " WHERE JSON_ARRAY_CONTAINS(JSON_EXTRACT(json, '$.neighbors'), 'foo1') != false"))
+                .skippingTypesCheck()
+                .matches(firstVendor)
+                .isFullyPushedDown();
+    }
+
+    @Test
+    public void testNotJsonArrayContains()
+    {
+        String otherVendors = "VALUES 'vendor2','vendor3','vendor4','vendor5','vendor6','vendor7'";
+        assertThat(query("SELECT vendor FROM " + JSON_TABLE + " WHERE NOT(JSON_ARRAY_CONTAINS(JSON_EXTRACT(json, '$.neighbors'), 'foo1'))"))
+                .skippingTypesCheck()
+                .matches(otherVendors)
+                .isFullyPushedDown();
+    }
+
+    @Test
+    public void testJsonContainsEquals()
+    {
+        // equals false case
+        String otherVendors = "VALUES 'vendor2','vendor3','vendor4','vendor5','vendor6','vendor7'";
+        assertThat(query("SELECT vendor FROM " + JSON_TABLE + " WHERE CONTAINS(ARRAY['vendor1'], JSON_EXTRACT_SCALAR(json, '$.vendor')) = false"))
+                .skippingTypesCheck()
+                .matches(otherVendors)
+                .isFullyPushedDown();
+
+        // equals true case
+        String firstVendor = "VALUES 'vendor1'";
+        assertThat(query("SELECT vendor FROM " + JSON_TABLE + " WHERE CONTAINS(ARRAY['vendor1'], JSON_EXTRACT_SCALAR(json, '$.vendor')) = true"))
+                .skippingTypesCheck()
+                .matches(firstVendor)
+                .isFullyPushedDown();
+
+        // not equals true case
+        assertThat(query("SELECT vendor FROM " + JSON_TABLE + " WHERE CONTAINS(ARRAY['vendor1'], JSON_EXTRACT_SCALAR(json, '$.vendor')) != true"))
+                .skippingTypesCheck()
+                .matches(otherVendors)
+                .isFullyPushedDown();
+
+        // not equals false case
+        assertThat(query("SELECT vendor FROM " + JSON_TABLE + " WHERE CONTAINS(ARRAY['vendor1'], JSON_EXTRACT_SCALAR(json, '$.vendor')) != false"))
+                .skippingTypesCheck()
+                .matches(firstVendor)
+                .isFullyPushedDown();
+    }
+
+    @Test
+    public void testNotJsonContains()
+    {
+        String otherVendors = "VALUES 'vendor2','vendor3','vendor4','vendor5','vendor6','vendor7'";
+        assertThat(query("SELECT vendor FROM " + JSON_TABLE + " WHERE NOT(CONTAINS(ARRAY['vendor1'], JSON_EXTRACT_SCALAR(json, '$.vendor')))"))
+                .skippingTypesCheck()
+                .matches(otherVendors)
+                .isFullyPushedDown();
+    }
+
+    @Test
+    public void testEqualsAndJsonExtractIsNull()
+    {
+        assertThat(query("SELECT vendor FROM " + JSON_TABLE + " WHERE vendor = 'vendor1' AND JSON_EXTRACT(json, '$.selector') IS NULL"))
+                .skippingTypesCheck()
+                .matches("VALUES 'vendor1'")
```

Comment:
Remove `skippingTypesCheck` and change this query to `VALUES VARCHAR 'vendor1'`.

Reply:
fixed